server/src/main/java/org/opensearch/index/compositeindex/CompositeIndexMetadata.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex;
+
+import org.opensearch.index.mapper.CompositeMappedFieldType;
+
+/**
+ * This class represents the metadata of a Composite Index, which includes information about
+ * the composite field name, type, and the specific metadata for the type of composite field
```

Comment:
nit : Specific metadata such as tree metadata is not present now, should we update the javadoc ?